### PR TITLE
Show filename on share link download

### LINKED ISSUE #5475: Add the filename to the share link
When sharing a file by link it maybe is useful to add the filename to the link itself. Just to better identify what is behind this link. I think it doesn't even need to have an impact on the actual sharing, but should just help to identify the content if you only have the links.

Something like 

* https://cloud.example.org/s/lc3cANiTLMXfgzo#test.avi
* https://cloud.example.org/s/lc3cANiTLMXfgzo#expenses.docx

@nextcloud/designers @nextcloud/sharing What do you think?

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/5475

- Folder share: https://dev.domain.com/s/67T62nzJ5qpt35N/download
- File share: https://dev.domain.com/s/67T62nzJ5qpt35N/download/100MB.tar.gz

### CODE REVIEW

File: apps/files_sharing/appinfo/routes.php [hunk 1]
```diff
@@ -50,9 +50,10 @@
 		],
 		[
 			'name' => 'Share#downloadShare',
-			'url' => '/s/{token}/download',
+			'url' => '/s/{token}/download/{filename}',
 			'verb' => 'GET',
 			'root' => '',
+			'defaults' => ['filename' => '']
```
Comment (MorrisJobke) [inline apps/files_sharing/appinfo/routes.php:56]:
```suggestion
			'defaults' => ['filename' => ''],
```

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by juliusknorr — APPROVED** [review]
Nice :+1:

**Comment by juliushaertl** [comment]
Rebased and fixed the php codestyle isssue

**Comment by MorrisJobke** [comment]
20 is branched off -> merge

**Comment by jsalatiel** [comment]
Will it be backported to 19 and 18 ?

**Comment by szaimen** [comment]
> Will it be backported to 19 and 18 ?

I don't think so. It is not even backported to NC20, afaik.

**Comment by skjnldsv** [comment]
No it will not

**Pull request merged**